pkg/devcontainer/crane/run.go

```diff
@@ -0,0 +1,96 @@
+package crane
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/loft-sh/log"
+)
+
+var (
+	craneSigningKey string
+)
+
+const (
+	PullCommand    = "pull"
+	DecryptCommand = "decrypt"
+
+	GitCrane = "git"
+
+	BinPath = "devpod-crane" // FIXME
+
+	tmpDirTemplate = "devpod-crane-*"
+)
+
+type Content struct {
+	Files map[string]string `json:"files"`
+}
+
+func IsAvailable() bool {
+	_, err := exec.LookPath(BinPath)
+	return err == nil
+}
+
+func runCommand(command string, args ...string) (string, error) {
+	cmd := exec.Command(BinPath, append([]string{command}, args...)...)
+
+	var outBuf, errBuf bytes.Buffer
+	cmd.Stdout = &outBuf
+	cmd.Stderr = &errBuf
+
+	err := cmd.Run()
+	if err != nil {
+		return "", fmt.Errorf("failed to execute command: %v, error: %w", errBuf.String(), err)
+	}
+
+	return outBuf.String(), nil
+}
+
+func PullConfigFromSource(configSource string, log log.Logger) (string, error) {
```

Comment:
docbock please for public function :pray:

Reply:
added 👍